sdgx/data_models/metadata.py

```diff
@@ -39,7 +39,7 @@ class Metadata(BaseModel):
     primary_keys is used to store single primary key or composite primary key
     """
 
-    column_list: Set[str] = set()
+    column_list: List[str] = []
```

Comment:
I'm actually not too keen on changing column_list to list, I'm worried about duplicate processing happening once there are duplicate keys.

One possible way to make it a list is to use @property with setter and getter.

Reply:
> I'm actually not too keen on changing column_list to list, I'm worried about duplicate processing happening once there are duplicate keys.
> 
> One possible way to make it a list is to use `@property` with setter and getter.

I believe it is necessary to maintain the order information of columns in metadata, and as you said `@property` should be a better solution.